lib/routes/devices-and-sessions.js

```diff
@@ -124,15 +124,20 @@ module.exports = (log, db, config, customs, push, devices) => {
 
         devices.upsert(request, sessionToken, payload)
           .then(function (device) {
-            reply(Object.assign({}, {
-              id: sessionToken && sessionToken.deviceId,
-              name: sessionToken && sessionToken.deviceName,
-              type: sessionToken && sessionToken.deviceType,
-              pushCallback: sessionToken && sessionToken.callbackURL,
-              pushPublicKey: sessionToken && sessionToken.callbackPublicKey,
-              pushAuthKey: sessionToken && sessionToken.callbackAuthKey,
-              pushEndpointExpired: sessionToken && sessionToken.callbackIsExpired
-            }, device))
+            // We must respond with the full device record,
+            // including any default values for missing fields.
+            reply(Object.assign({
+              // These properties can be picked from sessionToken or device as appropriate.
+              pushCallback: sessionToken.callbackURL,
+              pushPublicKey: sessionToken.callbackPublicKey,
+              pushAuthKey: sessionToken.callbackAuthKey,
+              pushEndpointExpired: sessionToken.callbackIsExpired
```

Comment:
We can fix in another PR if it is a problem but I just want to check something: it seems like these four might be returning the wrong value, in the case where one or more of them was updated in the payload data? Or am I missing something?

Reply:
My bad, I didn't notice the `Object.assign` in there. Ignore me.